dobson/utils.py

```diff
@@ -62,3 +68,33 @@ def get_mac_addresses():
 
         mac_addr = device[-18:].strip(' ').lower().replace(' ', ':')
         yield mac_addr
+
+
+def get_unknown_mac_addresses():
+    """
+    Yields MAC Addresses connected to the router which are _not_ in devices.json. Useful for
+    tagging new devices
+    """
+    for mac_addr in get_mac_addresses():
+        if mac_addr not in KNOWN_DEVICES:
+            yield mac_addr
+
+
+def add_device(mac_addr, data):
+    """Adds an entry for the specified Mac Address to devices.json and KNOWN_DEVICES
+    If the specified Mac Address already exists, returns False and does not update"""
+    mac_addr = mac_addr.lower()
+    if mac_addr in KNOWN_DEVICES:
+        return False
+
+    # update devices.json
+    with open(config['dobson']['KnownDevicesFile'], 'r') as f:
+        devices = json.loads(f.read())
```

Comment:
This is duplicated from lines 28 and 29, so I'd just make this into a separate method or something, maybe one that just updates `KNOWN_DEVICES` and then read from that as a source of truth?